internal/provider/webhook_data_source.go

```diff
@@ -0,0 +1,197 @@
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *sdk.SDK, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *WebhookDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data *WebhookResourceModel
+	var item types.Object
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &item)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(item.As(ctx, &data, basetypes.ObjectAsOptions{
+		UnhandledNullAsEmpty:    true,
+		UnhandledUnknownAsEmpty: true,
+	})...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	id := data.ID.ValueString()
+	displayName := data.DisplayName.ValueStringPointer()
+	if id == "" && (displayName == nil || *displayName == "") {
+		resp.Diagnostics.AddError("either id or displayName must be set", "")
+		return
+	}
+
+	// If we have an ID, we can use the Get API to fetch the attribute
+	if id != "" {
+		request := operations.C1APIWebhooksV1WebhooksServiceGetRequest{
+			ID: id,
+		}
+		res, err := r.client.Webhooks.Get(ctx, request)
+		if err != nil {
+			resp.Diagnostics.AddError("failure to invoke API", err.Error())
+			return
+		}
+		if res == nil {
+			resp.Diagnostics.AddError("unexpected response from API", fmt.Sprintf("%v", res))
+			return
+		}
+		if res.StatusCode != 200 {
+			resp.Diagnostics.AddError(fmt.Sprintf("unexpected response from API. Got an unexpected response code %v", res.StatusCode), debugResponse(res.RawResponse))
+			return
+		}
+		if res.WebhooksServiceGetResponse == nil {
+			resp.Diagnostics.AddError("unexpected response from API. No response body", debugResponse(res.RawResponse))
+			return
+		}
+		data.RefreshFromSharedWebhook(res.WebhooksServiceGetResponse.Webhook)
+
+		// Save updated data into Terraform state
+		resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+		return
+	}
+
+	// If we don't have an ID but we do have a value we can use the Search API to fetch the latest data.
+	request := &shared.WebhooksSearchRequest{
+		DisplayName: displayName,
+	}
+	res, err := r.client.WebhooksSearch.Search(ctx, request)
+	if err != nil {
+		resp.Diagnostics.AddError("failure to invoke API", err.Error())
+		return
+	}
+	if res == nil {
+		resp.Diagnostics.AddError("unexpected response from API", fmt.Sprintf("%v", res))
+		return
+	}
+	if res.StatusCode != 200 {
+		resp.Diagnostics.AddError(fmt.Sprintf("unexpected response from API. Got an unexpected response code %v", res.StatusCode), debugResponse(res.RawResponse))
+		return
+	}
+	if res.WebhooksSearchResponse == nil {
+		resp.Diagnostics.AddError("unexpected response from API. No response body", debugResponse(res.RawResponse))
+		return
+	}
+
+	if len(res.WebhooksSearchResponse.List) == 0 {
+		resp.Diagnostics.AddError("unexpected response from API. Webhook was not found", debugResponse(res.RawResponse))
+		return
+	}
+
+	if len(res.WebhooksSearchResponse.List) > 2 {
```

Comment:
\>= 2 or > 1

Reply:
Oops I've been copy and pasting this wrong in a bunch of places, yikes

Reply:
fixed